File: t-tasks/src/main/java/com/teo/ttasks/data/local/WidgetHelper.java
package com.teo.ttasks.data.local;

import android.appwidget.AppWidgetManager;
import android.content.ComponentName;
import android.content.Context;

import com.teo.ttasks.R;
import com.teo.ttasks.widget.TasksWidgetProvider;

public class WidgetHelper {

    private final Context context;

    public WidgetHelper(Context applicationContext) {
        context = applicationContext;
    }

    /**
     * Update all the widgets
     *
     * @param taskListId task list identifier
     */
    public void updateWidgets(String taskListId) {
        final AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(context);
        int[] ids = appWidgetManager.getAppWidgetIds(new ComponentName(context, TasksWidgetProvider.class));
        PrefHelper prefHelper = new PrefHelper(context);
        for (int id : ids) {
            if (taskListId.equals(prefHelper.getWidgetTaskListId(id)))
                appWidgetManager.notifyAppWidgetViewDataChanged(id, R.id.widget_task_list);
        }
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/data/local/WidgetHelper.kt
package com.teo.ttasks.data.local

import android.appwidget.AppWidgetManager
import android.content.ComponentName
import android.content.Context

import com.teo.ttasks.R
import com.teo.ttasks.widget.TasksWidgetProvider

class WidgetHelper(private val context: Context, private val prefHelper: PrefHelper) {

    /**
     * Update all the widgets

     * @param taskListId task list identifier
     */
    fun updateWidgets(taskListId: String) {
        val appWidgetManager = AppWidgetManager.getInstance(context)
        val ids = appWidgetManager.getAppWidgetIds(ComponentName(context, TasksWidgetProvider::class.java))
        ids.filter { taskListId == prefHelper.getWidgetTaskListId(it) }
           .forEach { appWidgetManager.notifyAppWidgetViewDataChanged(it, R.id.widget_task_list) }
    }
}
